authentication/private_key_signer.go

```diff
@@ -20,15 +22,23 @@ type PrivateKeySigner struct {
 	keyFingerprint          string
 	algorithm               string
 	accountName             string
+	userName                string
 	hashFunc                crypto.Hash
 
 	privateKey *rsa.PrivateKey
 }
 
-func NewPrivateKeySigner(keyFingerprint string, privateKeyMaterial []byte, accountName string) (*PrivateKeySigner, error) {
-	keyFingerprintMD5 := strings.Replace(keyFingerprint, ":", "", -1)
+type PrivateKeySignerInput struct {
+	KeyFingerPrint     string
```

Comment:
I missed this a minute ago, @stack72 .  This member should be renamed to be `KeyFingerprint` or even simplified down to just `KeyID`.